apps/src/templates/AppView.jsx

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var ProtectedStatefulDiv = require('./ProtectedStatefulDiv.jsx');
+
+/**
+ * Top-level React wrapper for our standard blockly apps.
+ */
+var AppView = React.createClass({
+  propTypes: {
+    renderCodeApp: React.PropTypes.func.isRequired,
+    onMount: React.PropTypes.func.isRequired
+  },
+
+  componentDidMount: function () {
+    this.refs.codeApp.getDOMNode().innerHTML = this.props.renderCodeApp();
+    this.props.onMount();
+  },
+
+  render: function () {
+    return <ProtectedStatefulDiv ref="codeApp" />;
+  }
```

Comment:
One problem with the `hasStatefulChildren` approach is that if you forget to render the child which is a `ProtectedStatefulDiv`, then the component won't have stateful children :/ I see now that the wrapping solution I proposed has the problem that a developer can still forget to wrap their component.

I'm open to brainstorming further, but your investigation of the code/design example leads me to believe that we're likely to catch the problem of forgetting to initially render a stateful child in an understandable way, so I'm inclined to not to block merging of this PR on further brainstorming.